pkg/logging/klog.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Upbound Inc.
+// All rights reserved
+
+package logging
+
+import (
+	"flag"
+	"os"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"k8s.io/klog/v2"
+)
+
+// SetFilteredKlogLogger sets log as the logger backend of klog, but filtering
+// aggressively to avoid noise.
+func SetFilteredKlogLogger(log logr.Logger) {
```

Comment:
Perhaps this should have something WRT "request throttling" in the name, to make it obvious _why_ it's filtering. Unless you think we might include more in future?

Reply:
Was thinking that maybe we will add more some day, i.e. this is more like the standard klog setup for crossplane components.